Utils/test_upload_flow/create_test_branch.py

```diff
@@ -1,150 +1,324 @@
 import argparse
+import json
+import os
 import shutil
-from git import GitCommandError, Repo, Head
-from pathlib import Path
 import subprocess
-import json
+import time
+from pathlib import Path
+from typing import Union
+from git import GitCommandError, Head, Repo
 
+from Tests.scripts.utils import logging_wrapper as logging
+from Tests.scripts.utils.log_util import install_logging
+
+versions_dict = {}
+pack_items_dict = {}
 changed_packs = set()
 
 
+# HELPER FUNCTIONS
+
+
+def json_write(file_path: str, data: Union[list, dict]):
+    """ Writes given data to a json file
+
+    Args:
+        file_path: The file path
+        data: The data to write
+
+    """
+    with open(file_path, "w") as f:
+        f.write(json.dumps(data, indent=4))
+
+
+def get_pack_content_paths(pack_path: Path, marketplace='xsoar'):
+    """
+    Gets a dict of all the paths of the given pack content items as it is in the bucket.
+
+    Args:
+        pack_path (Path): The pack path.
+
+    Returns:
+        dict: The content items paths dict.
+    """
+    create_artifacts_command = ['demisto-sdk', 'create-content-artifacts', '-a',
+                                '.', "-p", f'{pack_path.name}', '--no-zip', '--packs']
+    if marketplace == 'marketplacev2':
```

Comment:
```suggestion
    if marketplace != 'xsoar':
```
we might add xpanse in the future

Reply:
done